modules/sdk-lib-mpc/src/tss/ecdsa/rangeproof.ts

```diff
@@ -241,10 +277,16 @@ export async function prove(
   hash.update(bigIntToBufferBE(w, modulusBytes));
   hash.update('$');
   const e = bigIntFromBufferBE(hash.digest()) % q;
+  // Note: steps for migration
+  //     1. upgrade SDK to this, producing old challenge but ready to verify proof with both old and new challenge
+  //     2. upgrade HSM, producing new challenge and ready to verify proof with both old and new challenge
+  //     3. upgrade SDK to the next version that produces new challenge (possibly verify proof with only new challenge)
+  // const { nonce, e } = rangeProofChallenge(curve, modulusBytes, pk, c, z, u, w);
```

Comment:
So sounds like the prq-requisite to merging this SDK pr is to first have the HSM and WP ready to support both old/new challenge and challenge proof values.

Should this PR be a draft for now?

Reply:
Well, this PR is ready for step 1 mentioned in the notes, upgrading to it shouldn't break anything but SDK will be ready for step 2.  The commented out code and disabling of eslint warnings are the things we need to enable in step 3.

Reply:
Agreed to make it draft for now, because a number of other changes that go with this proof in a single call to HSM are not settled.  See https://github.com/BitGo/hsm-firmware/pull/812/files#diff-d61bc8c0ab7f3b380b8be40aaac80278f55d32402ae71244dc774ab9754b6b8aR1263